### ISSUE TITLE
Make modulefiles work on Mac

### ISSUE THREAD

**Issue by sawenzel** [alisw/alidist#324]
The module files do not work on the MAC as they often set `LD_LIBRARY_PATH` but not `DYLD_LIBRARY_PATH`.

[ We could possibly fix this automatically when we generate those files ]

**Comment by sawenzel** [comment]
As discussed, a possible fix is to generate the Modulefiles based on an environment variable
`LIBENVNAME` (or similar) which will be set from AliBuild.

**Comment by ktf** [comment]
Yes, agreed. This depends on alisw/alibuild#218 to be implemented first.

**Comment by dberzano** [comment]
As discussed with @ktf, the proposed solution would be not to modify `aliBuild` but to add in each modulefile something like:

```
$([[ ${ARCHITECTURE:0:3} == osx ]] && echo 'prepend-path DYLD_LIBRARY_PATH $::env(FOO_ROOT)/lib')
```

To be noted that in some cases we need both `LD_LIBRARY_PATH` and `DYLD_LIBRARY_PATH` on OSX, so the `DYLD_LIBRARY_PATH` would be exported in addition to the former, and it will not replace it.

**Comment by ktf** [comment]
Agreed. This approach will not require changes in alibuild, which should address @sawenzel concerns.

**Comment by sawenzel** [comment]
No concern about that ( only: do we know if which cases we need both variables ? )

**Comment by dberzano** [comment]
This has been addressed in #358. I honestly do not remember which are the cases. As said in the commit's comment, however, we should not need at all any library path on OSX. This commit provides an immediate fix.
